src/Admin/AdminNotice.php

```diff
@@ -6,29 +6,60 @@
  */
 class AdminNotice {
     public static function init() {
-        add_action( 'admin_init', array( __CLASS__, 'check_business_profile_option' ) );
+        add_action('admin_init', array(__CLASS__, 'check_business_profile_option'));
+        add_action('wp_ajax_generate_business_profile_data', array(__CLASS__, 'handle_generate_business_profile_data'));
     }
 
     // Method to check if bpr_business_profile option exists
     public static function check_business_profile_option() {
         // Check if we are on the admin side
-        if ( is_admin() ) {
+        if (is_admin()) {
             // Check if the bpr_business_profile option exists
-            if ( ! get_option( 'bpr_business_profile' ) ) {
+            if (!get_option('bpr_business_profile')) {
                 // If option doesn't exist, display notice
-                add_action( 'admin_notices', array( __CLASS__, 'display_contact_notice' ) );
+                add_action('admin_notices', array(__CLASS__, 'display_contact_notice'));
             }
         }
     }
 
+    public static function generate_business_profile_data() {
+        // Example logic to generate business profile data from website settings
+        $site_name = get_option('blogname');
+        $site_description = get_option('blogdescription');
+        $admin_email = get_option('admin_email');
+
+        $generated_data = array(
+            'name' => $site_name,
+            'description' => $site_description,
+            'email' => $admin_email,
+            // Add other relevant settings as needed
+        );
+
+        update_option('bpr_business_profile', $generated_data);
+    }
+
     // Method to display notice if option doesn't exist
     public static function display_contact_notice() {
         ?>
         <div class="notice notice-error">
-            <p><?php _e( 'No data was found for your account. Please contact the Website Pro team.', 'business-profile-render' ); ?></p>
+            <p><strong>Business Profile Data Missing:</strong> It looks like your Business Profile Data is missing after the recent migration. If you need assistance, please contact the Website Pro Team.</p>
```

Comment:
Please change the notice-error text to "It seems your Business Profile Data is unavailable after the recent migration. To restore it, you can generate the data based on your business name and email. Simply click the button below. If you need further assistance, please reach out to the Website Pro Team."

Otherwise it is LGTM